skimage/filters/_gabor.py

```diff
@@ -1,48 +1,338 @@
+                psin = -x[w[n + step]] / r
+
+                # Base 2-dimensional rotation
+                A[w[n], w[n]] = pcos
+                A[w[n], w[n + step]] = -psin
+                A[w[n + step], w[n]] = psin
+                A[w[n + step], w[n + step]] = pcos
+                x[w[n + step]] = 0
+                x[w[n]] = r
+
+        R = np.matmul(A, R)  # Multiply R by current matrix of stage A
+
+    return R
+
+
+def _compute_rotation_matrix(src, dst, use_homogeneous_coords=False):
+    """Generates a matrix for the rotation of one vector to the direction
+    of another.
+
+    Parameters
+    ----------
+    src : (N, ) array
+        Vector to rotate.
+    dst : (N, ) array
+        Vector of desired direction.
+    use_homogeneous_coords : bool, optional
+        If the input vectors should be treated as homoegeneous coordinates.
+
+    Returns
+    -------
+    M : (N, N) array
+        Matrix that rotates `src` to coincide with `dst`.
+
+    References
+    ----------
+    .. [1] Ognyan Ivanov Zhelezov. One Modification which Increases Performance
+           of N-Dimensional Rotation Matrix Generation Algorithm. International
+           Journal of Chemistry, Mathematics, and Physics, Vol. 2 No. 2, 2018:
+           pp. 13-18. https://dx.doi.org/10.22161/ijcmp.2.2.1
+
+    Examples
+    --------
+    >>> X = np.asarray([1, 0])
+    >>> Y = np.asarray([0.5, 0.5])
+
+    >>> M = _compute_rotation_matrix(X, Y)
+    >>> Z = np.matmul(M, X)
+
+    >>> uY = Y / np.linalg.norm(Y)
+    >>> np.allclose(Z, uY)
+    True
+    """
+    homogeneous_slice = -use_homogeneous_coords or None
+    X = _normalize(np.array(src)[:homogeneous_slice])
+    Y = _normalize(np.array(dst)[:homogeneous_slice])
+
+    if use_homogeneous_coords:
+        X = np.append(X, 1)
+        Y = np.append(Y, 1)
+
+    w = np.flatnonzero(~np.isclose(X, Y))  # indices of difference
+
+    Mx = _compute_projection_matrix(X, w)
+    My = _compute_projection_matrix(Y, w)
+    M = np.matmul(My.T, Mx)
+
+    return M
+
+
+def _decompose_quasipolar_coords(r, thetas):
+    """Decomposes quasipolar coordinates into their cartesian components.
+
+    Quasipolar coordinate decomposition is defined as follows:
+
+    .. math::
+
+         \left\{
+         \begin{array}{llllll}
+                 x_0     & \quad = r \sin \theta_0 \sin \theta_1 ... \sin \theta_{n-1} \\
+                 x_1     & \quad = r \cos \theta_0 \sin \theta_1 ... \sin \theta_{n-1} \\
+                 x_2     & \quad = r \cos \theta_1 \sin \theta_2 ... \sin \theta_{n-1} \\
+                 ...                                                                   \\
+                 x_{n-1} & \quad = r \cos \theta_{n-2} \sin \theta_{n-1}               \\
+                 x_n     & \quad = r \cos \theta_{n-1}
+         \end{array}
+         \right.
+
+    Parameters
+    ----------
+    r : float
+        Radial coordinate.
+    thetas : (N, ) array
+        Quasipolar angles.
```

Comment:
Some explanation of where these angles lie would be helpful (given the "quasi" nature you explained to me).